cmd/k8s-operator/sts.go

```diff
@@ -93,8 +92,6 @@ const (
 	podAnnotationLastSetClusterDNSName    = "tailscale.com/operator-last-set-cluster-dns-name"
 	podAnnotationLastSetTailnetTargetIP   = "tailscale.com/operator-last-set-ts-tailnet-target-ip"
 	podAnnotationLastSetTailnetTargetFQDN = "tailscale.com/operator-last-set-ts-tailnet-target-fqdn"
-	// podAnnotationLastSetConfigFileHash is sha256 hash of the current tailscaled configuration contents.
-	podAnnotationLastSetConfigFileHash = "tailscale.com/operator-last-set-config-file-hash"
```

Comment:
IIUC old proxies (i.e. pinned to 1.76) running with the latest operator (1.78) won't react to config changes. Some options:

1. Leave it as-is
2. Stagger the changes so we watch config files first and then remove config hash annotations in a few releases
3. Observe capability version or some kind of configurable option so that we can _optionally_ remove config hash annotations if the proxy is new enough to react to changes on its own.

TBH I don't really like any of those options though 😅 WDYT?

Reply:
Good catch

We added support for proxies writing capver to the state Secret in 1.78, so I think it makes sense to utilize that and set the configfile hash if the capver is older than 110 (and set capver to 110 in this PR).
We'll only need to maintain that check till 1.84

Reply:
> Leave it as-is

We can't leave it as is especially for egress ProxyGroup- when I was working on failover improvements for ProxyGroup I discovered that we practically don't have a way to ensure smooth failover in all cases - so should aim for less restarts

Reply:
I've changed this so that:
- previous logic is preserved for all non ProxyGroup proxies + egress proxies below capver 110
- proxies where the capver can be determined on updates preserve the annotation at whatever value they currently have it set (to prevent unnecessary updates/loops caused by removing it)

capver 110 should be good enough since this was bumped ~4 weeks ago and we haven't released since apart from the one unstable I cut this morning

I've tested this quite extensively with capver above/below 110 and upgrades/downgrades